### PR TITLE
Fix client update time and update height for batch consensus updates

### PR DESCRIPTION
**Pull request by Wizdave97**

### CODE REVIEW

File: modules/src/core/ics03_connection/handler/verify.rs [hunk 1]
```diff
@@ -56,6 +57,43 @@ pub fn verify_proofs<HostFunctions: HostFunctionsProvider>(
     }
 }
 
+#[cfg(feature = "skip_host_consensus_verification")]
+/// Entry point for verifying all proofs bundled in any ICS3 message.
+pub fn verify_proofs<HostFunctions: HostFunctionsProvider>(
```
Comment (seunlanlege) [inline modules/src/core/ics03_connection/handler/verify.rs:62]: isn't this unused

File: modules/src/core/ics02_client/context.rs [hunk 2]
```diff
@@ -117,21 +128,20 @@ pub trait ClientKeeper {
...
+                                self.store_update_time(
+                                    res.client_id.clone(),
+                                    height,
+                                    res.processed_time,
+                                )?;
+                                self.store_update_height(
+                                    res.client_id.clone(),
+                                    height,
+                                    res.processed_height,
+                                )?;
```
Comment (seunlanlege) [inline modules/src/core/ics02_client/context.rs:140]: don't we only want to store the most recent height?

File: modules/src/core/ics03_connection/handler/verify.rs (hunk 1 above)
Comment (Wizdave97) [reply modules/src/core/ics03_connection/handler/verify.rs:62]: No, this is the function used in verifying connection messages.

Comment (seunlanlege) [reply modules/src/core/ics03_connection/handler/verify.rs:62]: oh ok

File: modules/src/core/ics02_client/context.rs (hunk 2 above)
Comment (Wizdave97) [reply modules/src/core/ics02_client/context.rs:140]: We need to be able to retrieve the time a client was updated for every height(as long as the consensus state for that height is still on chain that is) to adequately verify if the delay period was observed when processing packets

**Pull request merged**